docs/Style.thy

```diff
@@ -0,0 +1,260 @@
+(*
+     Copyright 2021, Data61, CSIRO (ABN 41 687 119 230)
+
+     SPDX-License-Identifier: CC-BY-SA-4.0
+*)
+
+chapter \<open>An Isabelle Syntax Style Guide\<close>
+
+theory Style
+  imports Main
+begin
+
+typedecl type_foo
+typedecl type_bar
+
+definition valid ::
+  "'a \<Rightarrow> 'b \<Rightarrow> 'c \<Rightarrow> bool" ("\<lbrace>_\<rbrace>/ _ /\<lbrace>_\<rbrace>")
+where
+  "\<lbrace>P\<rbrace> f \<lbrace>Q\<rbrace> \<equiv> undefined"
+
+definition
+  pred_conj :: "('a \<Rightarrow> bool) \<Rightarrow> ('a \<Rightarrow> bool) \<Rightarrow> ('a \<Rightarrow> bool)" (infixl "and" 35)
+where
+  "pred_conj P Q \<equiv> \<lambda>x. P x \<and> Q x"
```

Comment:
I vote for a separate helper theory. It'll be nicer when you can start reading actual content from the top.

Reply:
Is there a reason to not just import `Lib.NonDetMonad`? I guess it would mean adding a `ROOT` file with a `Docs` session, but I guess we would want that anyway, to avoid proliferation of unchecked Isabelle theories in the repo.

Reply:
I added a helper theory because it's easier and faster, I hope that's ok for now.

Reply:
A helper theory is fine from my side, but I don't see it popping up in the diff. Am I looking at the wrong version?

Reply:
Oh, I do see it, but it's not used in `Style.thy` yet. Can you please add that? Basically we want the theory to start with the text explanation (or at least as soon as practical).

Reply:
Oops, should be fixed now.